var/spack/repos/builtin/packages/ufs-weather-model/package.py

```diff
@@ -17,68 +17,160 @@ class UfsWeatherModel(CMakePackage):
     url = "https://github.com/ufs-community/ufs-weather-model/archive/refs/tags/ufs-v1.1.0.tar.gz"
     git = "https://github.com/ufs-community/ufs-weather-model.git"
 
-    maintainers("t-brown")
+    maintainers("AlexanderRichert-NOAA")
 
-    version(
-        "2.0.0",
-        tag="ufs-v2.0.0",
-        commit="e3cb92f1cd8941c019ee5ef7da5c9aef67d55cf8",
-        submodules=True,
-    )
-    version(
-        "1.1.0",
-        tag="ufs-v1.1.0",
-        commit="5bea16b6d41d810dc2e45cba0fa3841f45ea7c7a",
-        submodules=True,
-    )
+    version("develop", branch="develop", submodules=True)
+    version("2.0.0", tag="ufs-v2.0.0", submodules=True)
+    version("1.1.0", tag="ufs-v1.1.0", submodules=True)
 
+    variant("mpi", default=True, description="Enable MPI")
     variant(
         "32bit", default=True, description="Enable 32-bit single precision arithmetic in dycore"
     )
-    variant("avx2", default=False, description="Enable AVX2 instructions")
     variant(
-        "ccpp", default=True, description="Enable the Common Community Physics Package (CCPP))"
+        "ccpp_32bit",
+        default=False,
+        description="Enable CCPP_32BIT (single precision arithmetic in slow physics)",
+    )
+    variant("avx2", default=True, description="Enable AVX2 instruction set")
+    variant(
+        "simdmultiarch", default=False, description="Enable multi-target SIMD instruction sets"
```

Comment:
What architectures is this for supporting? This may also be better handled in the target.

Reply:
This is a cmake option that adds '-axSSE4.2,CORE-AVX2' to cmake cflags/fflags.

Reply:
Ok yes this should be combined with the previous and use the `target` -- you can always target an earlier architecture if you don't want those features

Reply:
You can do ranges of targets, so you could say `when="target=:haswell"` or whatever the appropriate target is. But you'd have to have a check for the intel cpus, a check for the amd cpus, and checks for ppc and arm.

Much better to just trigger it by whether the arch supports it and build for the older arch if you don't want to use it.

Reply:
see comment in other thread about setting target purely through spack